constellations/gemini-user/src/afh/afh.js

```diff
@@ -0,0 +1,126 @@
+"use strict";
+require("./afh.sass");
+import React from "react";
+import API from "../utils/api.js";
+import moment from "moment";
+
+export class AFHPage extends React.Component {
+    state = {
+        currentTab: "Math",
+        sessions: [],
+    };
+
+    componentDidMount() {
+        API.get("api/askforhelp/all").then((res) => {
+            const afh = res.data;
+            this.setState({
+                sessions: afh,
+            });
+        });
+    }
+
+    openSubject = (subjectName) => {
+        this.setState({
+            currentTab: subjectName,
+        });
+    };
+
+    render() {
+        let currentSub = this.state.sessions.filter(
+            (session) => session.subject == this.state.currentTab
+        );
+
+        let showSessions = currentSub.map((row, index) => {
+            return <AfhSessionRow key={index} row={row} />;
+        });
+
+        const subjectDisplayNames = {
+            math: "Math",
+            english: "English",
+            computer: "Computer Programming",
+        };
+
+        return (
+            <div id="view-afh">
+                <h1>Ask for Help</h1>
+                <div className="description">
+                    We provide free sessions for students to ask for additional
+                    assistance on any of our program subjects. Please fill the
+                    form to let us know you are coming. You must be registered
+                    with one of our programs to attend.
+                </div>
+
+                <h1>Ask for Help Sessions by Subject</h1>
+                <div className="tab">
+                    <TabButton
+                        currentTab={this.state.currentTab}
+                        openSubject={this.openSubject}
+                        subjectDisplayNames={this.subjectDisplayNames}
+                    />
+                </div>
+
+                <div className="showTab">{showSessions}</div>
+            </div>
+        );
+    }
+}
+
+class TabButton extends React.Component {
+    render() {
+        let currentTab = this.props.currentTab;
+        const subjectDisplayNames = this.props.subjectDisplayNames;
```

Comment:
let show = this.props.show; // If show is true, show the tab
// if it is false, then hide the tab

let subject = this.props.subject;
let displayName = subjectDisplayNames[subject]; // <= this will grab the value from `subjectDisplayNames` given the key `subject`. For example, `subjectDisplayNames["computer"]` will give us "Computer Programming"